core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
+    title: Enable factor reset notifications for end users 
+    impact: 80
+    docs: 
       desc: |
-        The super admin role has the highest permissions of all the admins in Okta. Super admins can create other admins, assign or remove permissions, and perform all other admin activities. Limiting the number of administrators with this level of permissions can reduce the odds of a bad actor gaining access to one of these highly privileged accounts.
-      audit: |
-        __Use `cnspec run`__
+        When enabled, end users are sent an email notification to inform them that one or more factors have been reset for their account.
 
-        Run this query:
+        ### Okta Recommends 
 
-          ```bash
-          cnspec run okta -c "okta.users.where(roles.all(type == "SUPER_ADMIN")).length <= 3" --organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        Enable this email notification to inform end users when one or more factors have been reset or removed.
 
-        __Use the cnspec shell__
+        ### Security impact 
 
-        1. Launch the cnspec shell:
+        High
 
-          ```bash
-          cnspec shell okta ---organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        ### User impact 
 
-        2. Run this query:
+        Low
 
-          ```mql
-          okta.users.where(roles.all(type == "SUPER_ADMIN")).length <= 3
-          ```
-      remediation: "### Use the Okta Admin Console\n\n1. Log into the organization using a super admin account.\n2. Select **Security > Administrators**. \n3. Select the **Admins** column and remove users with the **Super Administrator** role.\n"
-  - uid: mondoo-okta-security-okta-mfa-strong-factor
-    title: Use strong MFA factors
-    impact: 100
-    mql: |
-      okta.policies.accessPolicy.where(name == "MFA").all(rules.all(actions["appSignOn"]["verificationMethod"]["constraints"] { _["possession"]["deviceBound"] == "REQUIRED"} ))
+        End users are sent an email notification if they or an admin reset a factor for their account.
+
+    variants: 
+      - uid: mondoo-okta-security-factor-reset-notifications-for-end-users-runtime
+      - uid: mondoo-okta-security-factor-reset-notifications-for-end-users-terraform-hcl
+      - uid: mondoo-okta-security-factor-reset-notifications-for-end-users-terraform-plan
+  - uid: mondoo-okta-security-factor-reset-notifications-for-end-users-runtime
+    title: Enable factor reset notifications for end users 
+    impact: 80
+    filters: asset.platform == "okta-org"
+    mql: okta.organization.securityNotificationEmails['sendEmailForFactorResetEnabled'] == true
     docs:
+      remediation: 
+        - id: console 
+          desc: |
+            ### Configure factor reset notifications
+            1. In the Admin Console, go to **Security > General**.
+            2. Under **Security Notification Emails**, click **Edit**.
+            3. Set **MFA reset notification email** to **Enabled**.
+            4. Click **Save**.
```

Comment:
```suggestion
            4. Select **Save**.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.